test/assertions.js

```diff
@@ -363,3 +338,88 @@ should.Assertion.add('ExtendedDataset', function assertExtendedDataset() {
   this.obj.entities.should.be.a.Number();
   if (this.obj.lastEntity != null) this.obj.lastEntity.should.be.an.isoDate();
 });
+
+should.Assertion.add('EntityRootFields', function() {
+  this.params = { operator: 'have root level entity fields' };
+
+  this.obj.should.have.property('uuid').which.is.a.String();
+  this.obj.should.have.property('datasetName').which.is.a.String();
+  this.obj.should.have.property('createdAt').which.is.a.isoDate();
+  this.obj.should.have.property('updatedAt').which.is.nullOrIsoDate();
+  this.obj.should.have.property('deletedAt').which.is.nullOrIsoDate();
+});
+
+should.Assertion.add('Source', function() {
+  this.params = { operator: 'to be a Source' };
+
+  this.obj.should.have.property('type').which.is.a.String();
+  this.obj.should.have.property('id').which.is.a.String();
+});
+
+should.Assertion.add('EntityDefSummaryFields', function() {
```

Comment:
Was this `EntityDefSummaryFields` assertion made to be reused as an intermediate structure in the other assertions? Could you do more of a cascading assertion object approach instead? Using `this.obj.should.be.a.ParentAssertion()`? It might be shorter, easier to reason about, and easier to toggle certain properties on/off during preliminary development.

```
should.Assertion.add('Entity', function() {
  this.params = { operator: 'to be an Entity' };

  // root property / summary fields
  this.obj.should.have.property('uuid').which.is.a.String();
  this.obj.should.have.property('datasetName').which.is.a.String();
  this.obj.should.have.property('createdAt').which.is.a.isoDate();
  this.obj.should.have.property('updatedAt').which.is.nullOrIsoDate();
  this.obj.should.have.property('deletedAt').which.is.nullOrIsoDate();

  // current version
  this.obj.should.have.property('currentVersion').which.is.an.EntityDef();
});

should.Assertion.add('ExtendedEntity', function() {
  this.params = { operator: 'to be a Extended Entity' };

  this.obj.should.be.an.Entity();
  this.obj.creator.should.be.an.Actor();
});
```

(and a similar thing for the entityDef)

Reply:
I started with cascading approach but that was making me duplicate fields because EntitySummary can Extended as well and that would contain ExtendedEntityDefSummary. So I can't have a single EntitySummary assertion that could be extended by other assertions.

Reply:
Created a separate PR https://github.com/getodk/central-backend/pull/820